integration-tests/tests/src/tests/sync/app.ts

```diff
@@ -235,6 +236,54 @@ describe("App", () => {
       expect(this.app.currentUser).to.be.null;
     });
 
+    it("currentUser is available from an App listener", async function (this: Mocha.Context &
+      AppContext &
+      RealmContext) {
+      expect(this.app.currentUser).to.be.null;
+
+      const credentials = Realm.Credentials.anonymous();
+
+      const handle = createPromiseHandle();
+      const listener = () => {
+        expect(this.app.currentUser).to.not.be.null;
```

Comment:
I'm thinking it may be safer to check a property on the object to be sure we're not getting e.g. an empty object back.
```suggestion
        expect(this.app.currentUser?.id).to.be.a("string");
```

Reply:
Then checking it's an instance of `User` is probably even better 🙂

Reply:
💪